### PR TITLE
Document how to disable chroma telemetry

### PR DESCRIPTION
**Pull request by mmalohlava**
The change enables Heap telemetry in Gradio UI to be able to improve
user experience.
The default tracking app id point to Heap project `Main` and its `Development` environment.

For any H2O.ai deployment we should use tracking app id from Heap
project `Main` and environment `Production`. @mmalohlava or from
@mtanco.

The analytics DO NOT track any prompts, only user interactions in UI
(click on buttons, links).

### CODE REVIEW

**Comment by mmalohlava** [comment]
@johnygomez @mtanco pls check

**Review by pseudotensor — COMMENTED** [review]
Needs to be disabled by default and enabled via CLI option only.  Any keys should be removed from repo and passed on CLI.

**Review by pseudotensor — CHANGES_REQUESTED** [review]
1

File: src/gradio_themes.py [hunk 1]
```diff
@@ -221,11 +221,23 @@ def get_simple_title(title, description):
...
+    """
+
+def get_heap_js(heapAppId: str) -> str:
+    return ("""globalThis.window.heap=window.heap||[],heap.load=function(e,t){window.heap.appid=e,window.heap.config=t=t||{};var r=document.createElement("script");r.type="text/javascript",r.async=!0,r.src="https://cdn.heapanalytics.com/js/heap-"+e+".js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(r,a);for(var n=function(e){return function(){heap.push([e].concat(Array.prototype.slice.call(arguments,0)))}},p=["addEventProperties","addUserProperties","clearEventProperties","identify","resetIdentity","removeEventProperty","setEventProperties","track","unsetEventProperty"],o=0;o<p.length;o++)heap[p[o]]=n(p[o])};"""
+           f"""heap.load("{heapAppId}");""")
+
+def wrap_js_to_lambda(*args: str) -> str:
+    newline = "\n"
+    return f"""
+        () => {{
```
Comment (johnygomez) [inline src/gradio_themes.py:240]: nit: Why it needs to have double brackets?

File: src/gradio_runner.py [hunk 2]
```diff
@@ -2476,7 +2477,8 @@ def count_chat_tokens(model_state1, chat1, prompt_type1, prompt_dict1,
                        ,
                        queue=False, api_name='stop' if allow_api else None).then(clear_torch_cache, queue=False)
 
-        demo.load(None, None, None, _js=get_dark_js() if kwargs['dark'] else None)
+        app_js = wrap_js_to_lambda(get_dark_js(), get_heap_js(get_heap_app_id()))
+        demo.load(None, None, None, _js=app_js)
```
Comment (johnygomez) [inline src/gradio_runner.py:2481]: QQ: Just to double-check, when exactly is this script called on the UI side? Is it when the page is loaded? Note that the heap script is downloading and using additional scripts, which could be a problem if called in some weird way or sandboxed. I assume however it's tested and it's working.

**Review by johnygomez — APPROVED** [review]
LGTM, thanks Michal, I have just a few points to double-check.

File: src/gradio_themes.py (hunk 1 above)
Comment (mmalohlava) [reply src/gradio_themes.py:240]: string interpolation requires it - f-string "{}" is treated as a reference to variable.

**Comment by mmalohlava** [comment]
@pseudotensor 
> Needs to be disabled by default and enabled via CLI option only. Any keys should be removed from repo and passed on CLI.
the whole point is to collect telemetry to improve our understanding of end-users.

**Review by arnocandel — APPROVED** [review]

**Review by pseudotensor — APPROVED** [review]

**Pull request merged**